### PR TITLE
[tests-only][full-ci] add test to list multiple shares

### PR DESCRIPTION
**Pull request by prashant-gurung899**
## Description
This PR adds test to list multiple shares(more than 10).

## Related Issue

- Related to: https://github.com/owncloud/ocis/issues/9737
- Bug Report: https://github.com/owncloud/ocis/issues/9642

## How Has This Been Tested?

- Locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/bootstrap/Sharing.php [hunk 1]
```diff
@@ -1962,6 +1962,17 @@ public function userGetsAllSharesSharedByHimUsingTheSharingApi(string $user):voi
...
+	/**
+	 * @When user :user gets all the shared folders
+	 *
+	 * @param string $user
+	 *
+	 * @return void
+	 */
+	public function userGetsAllTheSharedFolders($user) {
+		$this->setResponse($this->getAllshares($user));
+	}
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/Sharing.php:1974]: let's not use Sharing context for new steps related to shares. we have to use SharingNG (graph endpoint)

File: tests/acceptance/features/apiSharingNgShareInvitation/shareInvitations.feature [hunk 2]
```diff
@@ -3198,3 +3198,34 @@ Feature: Send a sharing invitations
...
+  Scenario Outline: list multiple shares
+    Given using spaces DAV path
+    And user "Alice" has created folder "<folder-name>"
+    And user "Alice" has sent the following resource share invitation:
+      | resource        | <folder-name> |
+      | space           | Personal      |
+      | sharee          | Brian         |
+      | shareType       | user          |
+      | permissionsRole | Viewer        |
+    And user "Brian" has a share "<folder-name>" synced
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNgShareInvitation/shareInvitations.feature:3212]: The scenarios should be like this according to the bug report:
```feature
- share multiple files/folders (>10) to a user
- sharee lists the received shares
- check all shares are available 
```

**Review by saw-jan — CHANGES_REQUESTED** [review]

File: tests/acceptance/features/apiSharingNgShareInvitation/shareInvitations.feature [hunk 3]
```diff
@@ -3198,3 +3198,39 @@ Feature: Send a sharing invitations
     Then the HTTP status code should be "200"
     And for user "Brian" the space Shares should contain these entries:
       | textfile.txt |
+
+  @issue-9642
+  Scenario Outline: list multiple shares
+    Given using spaces DAV path
+    And user "Alice" has created folder "<resource>"
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNgShareInvitation/shareInvitations.feature:3205]:
```suggestion
    
    And user "Alice" has created the following folders
      | resource |
      | folder1  |
      | folder2  |
      ...
      | folder10 |
```

File: tests/acceptance/features/apiSharingNgShareInvitation/shareInvitations.feature [hunk 4]
```diff
@@ -3198,3 +3198,39 @@ Feature: Send a sharing invitations
...
+    And user "Alice" has uploaded file with content "lorem" to "textfile1.txt"
+    And user "Alice" has uploaded file with content "lorem" to "textfile2.txt"
+    And user "Alice" has uploaded file with content "lorem" to "textfile3.txt"
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNgShareInvitation/shareInvitations.feature:3208]:
```suggestion
    And user "Alice" has uploaded the following files with content "lorem epsum"
      | resource      |
      | textfile1.txt |
      | textfile2.txt |
      ...
      | textfile5.txt |
```

File: tests/acceptance/features/apiSharingNgShareInvitation/shareInvitations.feature [hunk 5]
```diff
@@ -3198,3 +3198,39 @@ Feature: Send a sharing invitations
...
+    And user "Alice" has sent the following resource share invitation:
+      | resource        | <resource>    |
+      | space           | Personal      |
+      | sharee          | Brian         |
+      | shareType       | user          |
+      | permissionsRole | Viewer        |
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNgShareInvitation/shareInvitations.feature:3214]:
```suggestion
    And user "Alice" has shared the following files from space "Personal" with user "Brian" and role "Viewer":
      | resource      |
      | textfile1.txt |
      | textfile2.txt |
      ...
      | textfile5.txt |
    And user "Alice" has shared the following folders from space "Personal" with user "Brian" and role "Editor":
      | resource |
      | folder1  |
      | folder2  |
      ...
      | folder10 |
```

File: tests/acceptance/features/apiSharingNgShareInvitation/shareInvitations.feature [hunk 6]
```diff
@@ -3198,3 +3198,39 @@ Feature: Send a sharing invitations
...
+    And user "Brian" has a share "<resource>" synced
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNgShareInvitation/shareInvitations.feature:3215]:
```suggestion
```

File: tests/acceptance/features/apiSharingNgShareInvitation/shareInvitations.feature [hunk 7]
```diff
@@ -3198,3 +3198,39 @@ Feature: Send a sharing invitations
...
+    When user "Brian" lists the shares shared with him using the Graph API
+    Then for user "Brian" the space Shares should contain these entries:
+      | <resource> |
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNgShareInvitation/shareInvitations.feature:3218]:
```suggestion
    When user "Brian" lists the shares shared with him using the Graph API
    Then the HTTP status code should be "200"
    And the json response should contain the following shares:
      | resource      |
      | folder1       |
      ...
      | folder10      |
      | textfile1.txt |
      ...
      | textfile5.txt |
```

**Review by saw-jan — CHANGES_REQUESTED** [review]
>The scenarios should be like this according to the bug report:
>```feature
>- share multiple files/folders (>10) to a user
>- sharee lists the received shares
>- check all shares are available 
>```

The scenario should look like this

File: tests/acceptance/features/apiSharingNgShareInvitation/shareInvitations.feature (hunk 7 above)
Comment (prashant-gurung899) [reply tests/acceptance/features/apiSharingNgShareInvitation/shareInvitations.feature:3218]: Looking into it. :+1:

File: tests/acceptance/features/apiSharingNgShareInvitation/shareInvitations.feature [hunk 8]
```diff
@@ -3198,3 +3198,39 @@ Feature: Send a sharing invitations
     Then the HTTP status code should be "200"
     And for user "Brian" the space Shares should contain these entries:
       | textfile.txt |
+
+  @issue-9642
+  Scenario Outline: list multiple shares
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNgShareInvitation/shareInvitations.feature:3203]: Also, this scenario should go to `apiSharingNg1/sharedWithMe.feature` file

Comment (prashant-gurung899) [reply tests/acceptance/features/apiSharingNgShareInvitation/shareInvitations.feature:3203]: okay. sure!

File: tests/acceptance/features/apiSharingNg1/sharedWithMe.feature [hunk 9]
```diff
@@ -5127,3 +5127,63 @@ Feature: an user gets the resources shared to them
       | resource      |
       | testfile.txt  |
       | FolderToShare |
+
+  @issue-9642
+  Scenario: list multiple shares
+    Given using spaces DAV path
+    And user "Alice" has created the following folders
+      | path     |
```
[the remark this answers is not in the document]
Comment (prashant-gurung899) [reply tests/acceptance/features/apiSharingNg1/sharedWithMe.feature:5135]: @nirajacharya2 the array isn't sliced in this step.

[the remark this answers is not in the document]
Comment (prashant-gurung899) [reply tests/acceptance/features/apiSharingNg1/sharedWithMe.feature:5135]: It's in this step:
``` feature
And user "Alice" has shared the following files from space "Personal" with user "Brian" and role "Viewer":
```

[the remark this answers is not in the document]
Comment (prashant-gurung899) [reply tests/acceptance/features/apiSharingNg1/sharedWithMe.feature:5135]: > It's in this step:
> 
> ```
> And user "Alice" has shared the following files from space "Personal" with user "Brian" and role "Viewer":
> ```



I will remove the slicing logic from here.

File: tests/acceptance/features/bootstrap/SharingNgContext.php [hunk 10]
```diff
@@ -1724,4 +1724,68 @@ public function checkIfShareExists(string $share, string $sharee, string $sharer
 	public function userShouldHaveShare(string $sharee, string $share, string $sharer): void {
 		$this->checkIfShareExists($share, $sharee, $sharer);
 	}
+
+	/**
+	 * @Given user :sharer has shared the following files from space :space with user :sharee and role :role:
+	 * @Given user :sharer has shared the following folders from space :space with user :sharee and role :role:
```
Comment (PrajwolAmatya) [inline tests/acceptance/features/bootstrap/SharingNgContext.php:1730]:
```suggestion
	 * @Given /^user "([^"]*)" has shared the following (files|folders) from space "([^"]*)" with user "([^"]*)" and role "([^"]*)"$/
```

File: tests/acceptance/features/apiSharingNg1/sharedWithMe.feature [hunk 11]
```diff
@@ -5127,3 +5127,61 @@ Feature: an user gets the resources shared to them
       | resource      |
       | testfile.txt  |
       | FolderToShare |
+
+  @issue-9642
+  Scenario: list multiple shares
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg1/sharedWithMe.feature:5132]:
```suggestion
  Scenario: list multiple shares (> 10 shares)
```

File: tests/acceptance/features/bootstrap/SharingNgContext.php [hunk 12]
```diff
@@ -1724,4 +1724,66 @@ public function checkIfShareExists(string $share, string $sharee, string $sharer
 	public function userShouldHaveShare(string $sharee, string $share, string $sharer): void {
 		$this->checkIfShareExists($share, $sharee, $sharer);
 	}
+
+	/**
+     * @Given /^user "([^"]*)" has shared the following (?:files|folders) from space "([^"]*)" with user "([^"]*)" and role "([^"]*)":$/
+	 *
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SharingNgContext.php:1730]:
```suggestion
	/**
         * @Given /^user "([^"]*)" has shared the following (?:files|folders) from space "([^"]*)" with user "([^"]*)" and role "([^"]*)":$/
	 *
```

File: tests/acceptance/features/bootstrap/SharingNgContext.php [hunk 13]
```diff
@@ -1724,4 +1724,66 @@ public function checkIfShareExists(string $share, string $sharee, string $sharer
...
+		$rows = $table->getRows();
+		foreach ($rows as $row) {
+			if (isset($row[0])) {
+				$shareData[] = [
+					'resource' => $row[0],
+					'space' => $space,
+					'sharee' => $sharee,
+					'shareType' => 'user',
+					'permissionsRole' => $role,
+				];
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SharingNgContext.php:1750]:
```suggestion
				$shareData = [
					'resource' => $row[0],
					'space' => $space,
					'sharee' => $sharee,
					'shareType' => 'user',
					'permissionsRole' => $role,
				];
				$response = $this->sendShareInvitation($sharer, $data);
				// check that the request was successful
```

File: tests/acceptance/features/bootstrap/SharingNgContext.php [hunk 14]
```diff
@@ -1724,4 +1724,66 @@ public function checkIfShareExists(string $share, string $sharee, string $sharer
...
+			}
+		}
+
+		foreach ($shareData as $data) {
+			$this->featureContext->setResponse(
+				$this->sendShareInvitation($sharer, $data)
+			);
+		}
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SharingNgContext.php:1758]: we must not use `setResponse` within a Given step
```suggestion
```

File: tests/acceptance/features/bootstrap/SharingNgContext.php [hunk 15]
```diff
@@ -1724,4 +1724,66 @@ public function checkIfShareExists(string $share, string $sharee, string $sharer
...
+				}
+			}
+		}
+
+		$expectedShares = $table->getColumn(0);
+
+		foreach ($expectedShares as $expectedShare) {
+			Assert::assertTrue(
+				\in_array($expectedShare, $resourceNames),
+				"The resource '$expectedShare' was not found in the JSON response."
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SharingNgContext.php:1785]:
```suggestion
				"The share '$expectedShare' was not found in the response."
```

**Review by saw-jan — COMMENTED** [review]
looking good :+1:

**Review by saw-jan — APPROVED** [review]

**Pull request merged**